scheduler/cloudformation/cloudformation.go

```diff
@@ -931,6 +968,101 @@ func (m *Scheduler) Run(ctx context.Context, app *scheduler.App, process *schedu
 		return fmt.Errorf("error calling RunTask: %v", err)
 	}
 
+	for _, f := range runResp.Failures {
+		return fmt.Errorf("error running task %s: %s", aws.StringValue(f.Arn), aws.StringValue(f.Reason))
+	}
+
+	task := runResp.Tasks[0]
+
+	if attached {
+		if err := m.attach(ctx, task, in, out); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// attach attaches to the given ECS task.
+func (m *Scheduler) attach(ctx context.Context, task *ecs.Task, in io.Reader, out io.Writer) error {
+	defer tryClose(out)
+
+	if a, _ := arn.Parse(aws.StringValue(task.TaskArn)); a != nil {
+		// TODO: This should really go to a STDERR stream instead. Putting this
+		// on stdout breaks redirection, but not including it leads to
+		// confusion.
+		fmt.Fprintf(out, "Attaching to %s...\n", a.Resource)
+	}
+
+	descContainerInstanceResp, err := m.ecs.DescribeContainerInstances(&ecs.DescribeContainerInstancesInput{
+		Cluster:            task.ClusterArn,
+		ContainerInstances: []*string{task.ContainerInstanceArn},
+	})
+	if err != nil {
+		return fmt.Errorf("error describing container instance (%s): %v", aws.StringValue(task.ContainerInstanceArn), err)
+	}
+
+	containerInstance := descContainerInstanceResp.ContainerInstances[0]
+	descInstanceResp, err := m.ec2.DescribeInstances(&ec2.DescribeInstancesInput{
+		InstanceIds: []*string{containerInstance.Ec2InstanceId},
+	})
+	if err != nil {
+		return fmt.Errorf("error describing ec2 instance (%s): %v", aws.StringValue(containerInstance.Ec2InstanceId), err)
+	}
+
+	ec2Instance := descInstanceResp.Reservations[0].Instances[0]
+
+	// Wait for the task to start running. It will stay in the
+	// PENDING state while the container is being pulled.
+	if err := m.ecs.WaitUntilTasksNotPending(&ecs.DescribeTasksInput{
+		Cluster: task.ClusterArn,
+		Tasks:   []*string{task.TaskArn},
+	}); err != nil {
+		return fmt.Errorf("error waiting for %s to transition from PENDING state: %s", aws.StringValue(task.TaskArn), err)
+	}
+
+	// Open a new connection to the Docker daemon on the EC2
+	// instance where the task is running.
+	d, err := m.NewDockerClient(ec2Instance)
+	if err != nil {
+		return fmt.Errorf("error establishing docker client connection to %s", aws.StringValue(ec2Instance.InstanceId))
```

Comment:
Actually, it does already show this, since it's output above the error (`Attaching to task/<task id>...`).